src/packages/solid/components/CardContent/CardContent.tsx

```diff
@@ -0,0 +1,183 @@
+   * applied to the CardContent, they must also be set on this component
+   */
+  inset?: NodeProps['children'];
+
+  /**
+   * component rendered below the CardContent
+   *
+   * component will be shown/hidden on focus unless shouldCollapse is false
+   *
+   * states are not forwarded to this component. if it should react to states
+   * applied to the CardContent, they must also be set on this component
+   */
+  bottom?: NodeProps['children'];
+
+  /**
+   * props passed to the child ProgressBar, rendered at the bottom of the component.
+   * ProgressBar will not appear if progressBar.progress is 0/falsy
+   */
+  progressBar?: Partial<ProgressBarProps> | undefined;
+
+  /**
+   * component rendered in right section of the CardContent on the top
+   *
+   * component will be shown/hidden on focus if shouldCollapse is true
+   *
+   * states are not forwarded to this component. if it should react to states
+   * applied to the CardContent, they must also be set on this component
+   */
+  metadataTop?: NodeProps['children'];
+
+  /**
+   * component rendered in right section of the CardContent on the bottom
+   *
+   * component will be shown/hidden on focus if shouldCollapse is true
+   *
+   * states are not forwarded to this component. if it should react to states
+   * applied to the CardContent, they must also be set on this component
+   */
+  metadataBottom?: NodeProps['children'];
+  /**
+   * sets the component's color palette
+   */
+  tone?: Tone;
+
+  padding?: number[];
+
+  style?: Partial<CardContentStyles>;
+}
+
+const CardContent: Component<CardContentProps> = props => {
+  const [isFocused, setIsFocused] = createSignal(false);
+  return (
+    <node
+      use:withPadding={
+        props.padding ??
+        styles.Container.tones[props.tone ?? styles.tone]?.padding ??
+        styles.Container.base.padding
+      }
+      {...props}
+      onFocus={() => setIsFocused(true)}
+      onBlur={() => setIsFocused(false)}
+      style={[
+        ...[props.style].flat(),
+        styles.Container.tones[props.tone ?? styles.tone],
+        styles.Container.base
+      ]}
+      forwardFocus={1}
+    >
+      {/* Left side of component- where Tile is */}
+      <Tile
+        {...props}
+        borderRadius={[
+          styles.Container.base.borderRadius,
+          !props.shouldCollapse || isFocused() ? 0 : styles.Container.base.borderRadius,
+          !props.shouldCollapse || isFocused() ? 0 : styles.Container.base.borderRadius,
+          styles.Container.base.borderRadius
+        ]}
+        style={[styles.TileContainer.tones[props.tone ?? styles.tone], styles.TileContainer.base]}
+        width={
+          props.width ??
+          styles.TileContainer.tones[props.tone ?? styles.tone]?.width ??
+          styles.TileContainer.base.width
+        }
+        height={
+          props.height ??
+          styles.TileContainer.tones[props.tone ?? styles.tone]?.height ??
+          styles.TileContainer.base.height
+        }
```

Comment:
i think we might actually not need to do all this fallback-ing since we've got the style object getting passed in. if `props.height` is nullish it should just fall back to whatever the Component.style value is, can we test that out?

Reply:
ugh